src/main/java/com/kakao/borrowme/product/ProductService.java

```diff
@@ -30,23 +33,58 @@ public Slice<ProductResponse.FindAllDTO> findAll(Long lastProductId, Pageable pa
         return responseDTOSlice;
     }
 
-    public ProductResponse.FindByIdDTO findById(Long id) {
-        Product product = productRepository.findById(id).orElseThrow(
-                () -> new Exception404("존재하지 않는 제품입니다. : "+id,"product_not_existed"));
+    public ProductResponse.FindByIdDTO findById(Long productId) {
+        Product product = productRepository.findById(productId).orElseThrow(
+                () -> new Exception404("존재하지 않는 제품입니다. : "+productId,"product_not_existed"));
 
-        ProductImage productImage = productImageRepository.findByProductId(id); // ProductImage 정보 조회
+        ProductImage productImage = productImageRepository.findByProductId(productId); // ProductImage 정보 조회
 
         Company company = companyRepository.findById(product.getCompany().getId()).orElseThrow(
-                () -> new Exception404("존재하지 않는 회사입니다. : "+id,"company_not_existed")); // Company 정보 조회
+                () -> new Exception404("존재하지 않는 회사입니다. : "+productId,"company_not_existed")); // Company 정보 조회
         return new ProductResponse.FindByIdDTO(product, productImage, company);
     }
 
-    public Slice<ProductResponse.FindAllDTO> searchProducts(Long lastProductId, String keyword, Pageable pageable) {
-        Slice<Product> productSlice = productRepository.searchProducts(lastProductId, keyword, pageable);
+    public Slice<ProductResponse.FindAllDTO> searchProduct(Long lastProductId, String keyword, Pageable pageable) {
+        Slice<Product> productSlice = productRepository.searchProduct(lastProductId, keyword, pageable);
         Slice<ProductResponse.FindAllDTO> responseDTOSlice = productSlice.map(product -> {
             ProductImage productImage = productImageRepository.findByProductId(product.getId());
             return new ProductResponse.FindAllDTO(product, productImage);
         });
         return responseDTOSlice;
     }
+
+    public ProductResponse.RentDTO rentProduct(Long productId, ProductRequest.RentDTO requestDTO) {
+        Product product = productRepository.findById(productId).orElseThrow(
+                () -> new Exception404("존재하지 않는 제품입니다. : " + productId, "product_not_existed"));
+
+        LocalDateTime startAt = requestDTO.getStartAt();
+        LocalDateTime endAt = requestDTO.getEndAt();
+
+        // 예약 종료일이 예약 시작일보다 빠른 경우 예외 처리
+        if (startAt.isAfter(endAt)) {
+            throw new Exception400("예약 종료일이 예약 시작일보다 빠릅니다. : " + endAt, "rent_incorrect_period");
+        }
+
+        // 선택된 날짜가 과거로 설정 경우 예외 처리
+        LocalDateTime now = LocalDateTime.now();
+        if (startAt.isBefore(now) || endAt.isBefore(now)) {
+            throw new Exception400("과거 날짜값은 대여할 수 없습니다. : " + endAt, "rent_past_period");
+        }
```

Comment:
이 부분은 LocalDate로 해야 오늘 대여하더라도 에러가 안날 것 같아요 !

```suggestion
        if (startAt.toLocalDate().isBefore(today) || endAt.toLocalDate().isBefore(today)) {
            throw new Exception400("과거 날짜값은 대여할 수 없습니다. : " + endAt, "rent_past_period");
        }
```

저도 에러가 나서 이런 방식으로 수정했습니다 !

추가적으로 예약종료일이 시작일보다 빠른 경우 처리도 혹시모르니 `startAt.toLocalDate().isAfter(endAt.toLocalDate())`
이걸로 수정했어요 !